src/Cafe/IOSU/legacy/iosu_fpd.cpp

```diff
@@ -1143,6 +1147,31 @@ namespace iosu
 				return FPResult_Ok;
 			}
 
+			nnResult CallHandler_UpdateCommentAsync(FPDClient* fpdClient, IPCIoctlVector* vecIn, uint32 numVecIn, IPCIoctlVector* vecOut, uint32 numVecOut)
+			{
+				std::unique_lock _l(g_fpd.mtxFriendSession);
+				if (numVecIn != 1 || numVecOut != 0)
+					return FPResult_InvalidIPCParam;
+				if (!g_fpd.nexFriendSession)
+					return FPResult_RequestFailed;
+				DeclareInputPtr(newComment, char16_t, (vecOut[0].size / 2), 0);
```

Comment:
`char16_t` should be `uint16be`
The way this reads the string from the buffer doesn't match IOSU behavior. Take a look at `CallHandler_UpdateGameMode` for the general pattern.

Reply:
I changed the code using `CallHandler_UpdateGameMode` as a reference, and it works fine